include/highfive/bits/H5PropertyList_misc.hpp

```diff
@@ -15,11 +15,55 @@
 
 namespace HighFive {
 
+namespace {
+inline hid_t convert_plist_type(Properties::Type propertyType) {
+    // The HP5_XXX are macros with function calls so we can't assign
+    // them as the enum values
+    switch(propertyType) {
+    case Properties::OBJECT_CREATE:
+        return H5P_OBJECT_CREATE;
+    case Properties::FILE_CREATE:
+        return H5P_FILE_CREATE;
+    case Properties::FILE_ACCESS:
+        return H5P_FILE_ACCESS;
+    case Properties::DATASET_CREATE:
+        return H5P_DATASET_CREATE;
+    case Properties::DATASET_ACCESS:
+        return H5P_DATASET_ACCESS;
+    case Properties::DATASET_XFER:
+        return H5P_DATASET_XFER;
+    case Properties::GROUP_CREATE:
+        return H5P_GROUP_CREATE;
+    case Properties::GROUP_ACCESS:
+        return H5P_GROUP_ACCESS;
+    case Properties::DATATYPE_CREATE:
+        return H5P_DATATYPE_CREATE;
+    case Properties::DATATYPE_ACCESS:
+        return H5P_DATATYPE_ACCESS;
+    case Properties::STRING_CREATE:
+        return H5P_STRING_CREATE;
+    case Properties::ATTRIBUTE_CREATE:
+        return H5P_ATTRIBUTE_CREATE;
+    case Properties::OBJECT_COPY:
+        return H5P_OBJECT_COPY;
+    case Properties::LINK_CREATE:
+        return H5P_LINK_CREATE;
+    case Properties::LINK_ACCESS:
+        return H5P_LINK_ACCESS;
+    default:
+            HDF5ErrMapper::ToException<PropertyException>(
+                "Unsupported property list type");
+    }
+    return -1;
```

Comment:
I wonder whether ToException should become a macro to avoid the spurious return -1 at the bottom.  Adding [[noreturn]] to the function declaration may also suppress the compilation warning but I need to test it (see https://en.cppreference.com/w/cpp/language/attributes/noreturn).

Reply:
In fact that works, but you have to be careful and place [[ noreturn ]] at the very beginning of the function declaration, otherwise you get a warning from gcc. I wonder whether we should add it anyway as it's C++11 specific.

Reply:
IMHO it's avoiding one slightly ugly thing with another (macros) where the ugliness gets hidden. I'd prefer to keep it obvious.